### PR TITLE
[SQL][DOCS][MINOR] Fix typos and wrong phrases in docs

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This PR intends to fix typos and phrases in the `/docs` directory. To find them, I run the Intellij typo checker.

### Why are the changes needed?

For better documents.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by maropu** [comment]
@HyukjinKwon @srowen

File: docs/sql-ref-ansi-compliance.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ The following subsections present behaviour changes in arithmetic operations, ty
 ### Arithmetic Operations
 
 In Spark SQL, arithmetic operations performed on numeric types (with the exception of decimal) are not checked for overflows by default.
-This means that in case an operation causes overflows, the result is the same that the same operation returns in a Java/Scala program (e.g., if the sum of 2 integers is higher than the maximum value representable, the result is a negative number).
+This means that in case an operation causes overflows, the result is the same with the corresponding operation in a Java/Scala program (e.g., if the sum of 2 integers is higher than the maximum value representable, the result is a negative number).
```
Comment (maropu) [inline docs/sql-ref-ansi-compliance.md]: https://github.com/apache/spark/pull/27489#discussion_r380022536

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 2]
```diff
@@ -260,7 +260,7 @@ ALTER TABLE dbx.tab1 PARTITION (a='1', b='2') SET LOCATION '/path/to/part/ways'
 -- SET SERDE/ SERDE Properties
 ALTER TABLE test_tab SET SERDE 'org.apache.hadoop.hive.serde2.columnar.LazyBinaryColumnarSerDe';
 
-ALTER TABLE dbx.tab1 SET SERDE 'org.apache.madoop' WITH SERDEPROPERTIES ('k' = 'v', 'kay' = 'vee')
+ALTER TABLE dbx.tab1 SET SERDE 'org.apache.hadoop' WITH SERDEPROPERTIES ('k' = 'v', 'kay' = 'vee')
```
Comment (HyukjinKwon) [inline docs/sql-ref-syntax-ddl-alter-table.md]: hahaha madoop

**Review by gengliangwang — APPROVED** [review]
LGTM, thanks for doing this!

**Review by HyukjinKwon — APPROVED** [review]

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 2 above)
Comment (maropu) [reply docs/sql-ref-syntax-ddl-alter-table.md]: yea, lol ;)

Comment (gengliangwang) [reply docs/sql-ref-syntax-ddl-alter-table.md]: .....

**Comment by gengliangwang** [comment]
Merging to master/3.0

**Comment by maropu** [comment]
Thanks, @gengliangwang and @HyukjinKwon !

**Pull request closed without merging**